lib/std/fmt/parse_float/parse.zig

```diff
@@ -107,6 +107,7 @@ fn parsePartialNumberBase(comptime T: type, stream: *FloatStream, negative: bool
     tryParseDigits(MantissaT, stream, &mantissa, info.base);
     var int_end = stream.offsetTrue();
     var n_digits = @intCast(isize, stream.offsetTrue());
+    if (n_digits >= 2 and std.mem.eql(u8, stream.slice[0..2], "0x")) n_digits -= 2;
```

Comment:
I think it might be enough to just do

```suggestion
    // The base being 16 implies a 0x prefix, which shouldn't be included in the digit count
    if (info.base == 16) n_digits -= 2;
```

Reply:
Good idea! I agree your solution is simpler. I've updated the PR to do this.